flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -685,13 +687,27 @@ public CompletableFuture<Acknowledge> submitTask(
                     jobManagerConnection.getClassLoaderHandle();
             PartitionProducerStateChecker partitionStateChecker =
                     jobManagerConnection.getPartitionStateChecker();
+            TernaryBoolean changelogEnabled = TernaryBoolean.FALSE;
+            try {
+                changelogEnabled =
+                        InstantiationUtil.readObjectFromConfig(
+                                taskInformation.getTaskConfiguration(),
+                                "enablechangelog",
+                                getClass().getClassLoader());
```

Comment:
I understand that the reason of not using `StreamConfig.ENABLE_CHANGE_LOG_STATE_BACKEND` constant is that it's in a different module (`flink-streaming-java`), not in `flink-runtime`.
and `flink-runtime` should not depend on `flink-streaming-java`.

However, the opposite is fine I think, `flink-streaming-java` can depend on `flink-runtime` (there were some plans to eventually merge them IIRC),
so the constant `ENABLE_CHANGE_LOG_STATE_BACKEND` could be moved to `flink-runtime`.

An obvious name would be `TaskConfig`, but it already exists (in `flink-runtime`) and is used for other purposes (other types of tasks).

So how about adding `TaskExecutionConfig` class to `flink-runtime`, package `org.apache.flink.runtime.taskexecutor`, and placing the constant+parsing there?

cc: @akalash, @XComp, @pnowojski

Reply:
That sounds reasonable in my opinion. Already better than having a hard-coded String representation here. 
I'm just wondering whether config-related stuff should be better of in `flink-core` which is also a dependency of `flink-stream-java`.

I'd imagine that @zentol might have an opinion here as well.

Reply:
I'm not sure, but I think the idea sometime ago was to make "streaming api" runtime free. There were also independent plans of merging streaming runtime and batch runtime (or basically getting rid of batch runtime with dropping `DataSet API`). All in all it seems to me like `flink-core` would be a better place for config options? Indeed it would be nice to hear what @zentol thinks about it, as I presume/hope he has a better knowledge about those issues.

Reply:
I would put it into `flink-core` personally; not sure if there is a huge benefit in spreading config options around.

Reply:
@rkhachatryan  I rebased https://github.com/apache/flink/pull/20160 and modified this part, could you please take a review?